packages/jest-snapshot/src/inline_snapshots.js

```diff
@@ -60,26 +49,50 @@ const saveSnapshotsForFile = (
 ) => {
   const sourceFile = fs.readFileSync(sourceFilePath, 'utf8');
 
-  // Resolve project configuration.
-  // For older versions of Prettier, do not load configuration.
-  const config = prettier.resolveConfig
-    ? prettier.resolveConfig.sync(sourceFilePath, {
-        editorconfig: true,
-      })
-    : null;
-
-  // Detect the parser for the test file.
-  // For older versions of Prettier, fallback to a simple parser detection.
-  const inferredParser = prettier.getFileInfo
-    ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
-    : (config && config.parser) || simpleDetectParser(sourceFilePath);
-
-  // Format the source code using the custom parser API.
-  const newSourceFile = prettier.format(sourceFile, {
-    ...config,
-    filepath: sourceFilePath,
-    parser: createParser(snapshots, inferredParser, babelTraverse),
-  });
+  let newSourceFile;
+  if (prettier) {
+    // Resolve project configuration.
+    // For older versions of Prettier, do not load configuration.
+    const config = prettier.resolveConfig
+      ? prettier.resolveConfig.sync(sourceFilePath, {
+          editorconfig: true,
+        })
+      : null;
+
+    // Detect the parser for the test file.
+    // For older versions of Prettier, fallback to a simple parser detection.
+    const inferredParser = prettier.getFileInfo
+      ? prettier.getFileInfo.sync(sourceFilePath).inferredParser
+      : (config && config.parser) || simpleDetectParser(sourceFilePath);
+
+    // Format the source code using the custom parser API.
+    newSourceFile = prettier.format(sourceFile, {
+      ...config,
+      filepath: sourceFilePath,
+      parser: createParser(snapshots, inferredParser, babelTraverse),
+    });
+  } else {
+    const plugins = /\.tsx?$/.test(sourceFilePath) ? ['typescript'] : undefined;
```

Comment:
This won't work for a lot of cases. You'd need to either:

1. Do what `babel-jest` does and load the project babel config: https://github.com/facebook/jest/blob/master/packages/babel-jest/src/index.js#L70
2. Do what `prettier` does and enable all language features: https://github.com/prettier/prettier/blob/ca43aad88a5853eebc47958b3f69b47df5d78782/src/language-js/parser-babylon.js#L11-L45

Reply:
I was wondering about this. I'll see if I can make option 1. I'll try to break this with some test cases in the meantime.

Reply:
Went for 2 in the end. If it's good enough for prettier it should be good enough here.

Reply:
we'll want to load the user config (same as https://github.com/babel/babel-eslint/pull/711) at some point, but if this works that's fine for now 🙂

Reply:
@SimenB sorry missed this comment - do you think this can go in as-is for now?